src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/validation.cs

```diff
@@ -420,6 +418,26 @@ private static object ShowValidationAdornerOperation(object arg)
             return null;
         }
 
+        private static void ShowValidationAdornerWhenAdornerSiteGetsVisible(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            var adornerSite = sender as UIElement;
+
+            if (adornerSite == null)
+            {
+                return;
+            }
+
+            adornerSite.IsVisibleChanged -= ShowValidationAdornerOnVisible;
```

Comment:
Was this intended to unsubscribe the handler added from ShowValidationAdornerHelper?
If so, shouldn't it unsubscribe `ShowValidationAdornerWhenAdornerSiteGetsVisible` instead of `ShowValidationAdornerOnVisible`?

Reply:
It should. Somehow my latest edits are lost. Will fix this tomorrow.

Reply:
Done.